backend/controllers/controller_kubernetes_resources.go

```diff
@@ -10,89 +10,67 @@ import (
-
-	params := mux.Vars(r)
-	resourceType := params["resourceType"]
-	resourceName := params["resourceName"]
+	setJSONContentType(w)
 
-	queryParams := r.URL.Query()
-	namespace := queryParams.Get("namespace")
+	resourceType := getResourceType(r)
+	resourceName := getResourceName(r)
+	namespace := getNamespace(r)
 
 	resource, err := cluster.GetResource(resourceType, namespace, resourceName)
 	if err != nil {
-		w.WriteHeader(int(err.Code))
-		json.NewEncoder(w).Encode(err)
+		writeJSONError(w, err)
 		return
 	}
 
-	w.WriteHeader(http.StatusOK)
-	json.NewEncoder(w).Encode(resource)
+	writeJSONResponse(w, http.StatusOK, resource)
 }
 
 func ListResourcesController(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+	setJSONContentType(w)
 
-	params := mux.Vars(r)
-	resourceType := params["resourceType"]
-
-	queryParams := r.URL.Query()
-	namespace := queryParams.Get("namespace")
+	resourceType := getResourceType(r)
+	namespace := getNamespace(r)
 
 	resources, err := cluster.ListResources(resourceType, namespace)
 	if err != nil {
-		w.WriteHeader(int(err.Code))
-		json.NewEncoder(w).Encode(err)
+		writeJSONError(w, err)
 		return
 	}
 
-	w.WriteHeader(http.StatusOK)
-	json.NewEncoder(w).Encode(resources)
+	writeJSONResponse(w, http.StatusOK, resources)
 }
 
 func CreateResourceController(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
-
-	params := mux.Vars(r)
-	resourceType := params["resourceType"]
+	setJSONContentType(w)
 
-	queryParams := r.URL.Query()
-	namespace := queryParams.Get("namespace")
+	resourceType := getResourceType(r)
+	namespace := getNamespace(r)
 
 	var resource models.ResourceDetails
-	jsonErr := json.NewDecoder(r.Body).Decode(&resource.ResourceDetails)
-	if jsonErr != nil {
-		w.WriteHeader(http.StatusBadRequest)
-		json.NewEncoder(w).Encode(models.ModelError{Code: 404, Message: "Invalid request body"})
+	if !decodeJSONBody(w, r, &resource.ResourceDetails) {
 		return
 	}
 
 	resource, err := cluster.CreateResource(resourceType, namespace, resource)
 	if err != nil {
 		fmt.Println(err)
-		w.WriteHeader(int(err.Code))
-		json.NewEncoder(w).Encode(err)
+		writeJSONError(w, err)
 		return
 	}
 
-	w.WriteHeader(http.StatusCreated)
-	json.NewEncoder(w).Encode(resource)
+	writeJSONResponse(w, http.StatusCreated, resource)
 }
 
 func DeleteClusterResourceController(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
-
-	params := mux.Vars(r)
-	resourceType := params["resourceType"]
-	resourceName := params["resourceName"]
+	setJSONContentType(w)
 
-	queryParams := r.URL.Query()
-	namespace := queryParams.Get("namespace")
+	resourceType := getResourceType(r)
+	resourceName := getResourceName(r)
+	namespace := getNamespace(r)
 
 	err := cluster.DeleteResource(resourceType, namespace, resourceName)
```

Comment:
For consistency you can change this function to return model.Status{} also, but not mandatory